lib/spack/spack/concretize.py

```diff
@@ -73,9 +73,7 @@ def concretize_develop(self, spec):
         if not dev_info:
             return False
 
-        path = dev_info['path']
-        path = path if os.path.isabs(path) else os.path.join(
-            env.path, path)
+        path = os.path.normpath(os.path.join(env.path, dev_info['path']))
```

Comment:
Is this relying on the fact that `os.path.join` returns the second argument if it is absolute?

Reply:
Yeah, should also work on Windows iiuc: https://docs.python.org/3/library/os.path.html#os.path.join